### ISSUE TITLE
pip install fails with inconsistent metadata

### ISSUE THREAD

**Issue by rjfarmer** [aCOSMIC/aCOSMIC#556]
pip --version
pip 21.2.3

python3 --version
Python 3.10.4

````
pip install cosmic-popsynth
Defaulting to user installation because normal site-packages is not writeable
Collecting cosmic-popsynth
  Downloading cosmic-popsynth-3.4.0.tar.gz (198 kB)
     |████████████████████████████████| 198 kB 7.8 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/05/84/e8c586a2ce16bc9f4f898b2e982194a409f75d1a7a3c7da1f174049fddf6/cosmic-popsynth-3.4.0.tar.gz#sha256=669bbbe91397a08ed094dfeda2ca63f4eb3ee7f47ea44022304cbf94ef2139bc (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>3.5, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/05/84/e8c586a2ce16bc9f4f898b2e982194a409f75d1a7a3c7da1f174049fddf6/cosmic-popsynth-3.4.0.tar.gz#sha256=669bbbe91397a08ed094dfeda2ca63f4eb3ee7f47ea44022304cbf94ef2139bc has inconsistent version: filename has '3.4.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-3.3.0.tar.gz (177 kB)
     |████████████████████████████████| 177 kB 8.7 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/a2/f1/31061b97f7fd25c0f55469aaddd064b1a2a0237c670ef8e43c5404d4ca51/cosmic-popsynth-3.3.0.tar.gz#sha256=7dfccaefc915a20f3eaec9d474863caa61b32ddd5b4e69fadb28d4d9ae8f9239 (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>3.5, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/a2/f1/31061b97f7fd25c0f55469aaddd064b1a2a0237c670ef8e43c5404d4ca51/cosmic-popsynth-3.3.0.tar.gz#sha256=7dfccaefc915a20f3eaec9d474863caa61b32ddd5b4e69fadb28d4d9ae8f9239 has inconsistent version: filename has '3.3.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-3.2.0.tar.gz (169 kB)
     |████████████████████████████████| 169 kB 15.4 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/59/bc/737dc7bf3b26a26b0bd6bf57370b5b499104b9c78bf5a5c62f38a90fcf48/cosmic-popsynth-3.2.0.tar.gz#sha256=0846b4a54abfc0cc8ce1d5d091b87c287d0c0d58b6fea6585bac09afa0a53beb (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/59/bc/737dc7bf3b26a26b0bd6bf57370b5b499104b9c78bf5a5c62f38a90fcf48/cosmic-popsynth-3.2.0.tar.gz#sha256=0846b4a54abfc0cc8ce1d5d091b87c287d0c0d58b6fea6585bac09afa0a53beb has inconsistent version: filename has '3.2.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-3.1.0.tar.gz (258 kB)
     |████████████████████████████████| 258 kB 13.2 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/60/dd/76a25519612ebbfbe494e95bbfaaf0d281750d60edf5ef5eb0291b231714/cosmic-popsynth-3.1.0.tar.gz#sha256=6b87722d17e52c38ca54571bf79d45f477e97d58bfe7d63610a66f2534e1f344 (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/60/dd/76a25519612ebbfbe494e95bbfaaf0d281750d60edf5ef5eb0291b231714/cosmic-popsynth-3.1.0.tar.gz#sha256=6b87722d17e52c38ca54571bf79d45f477e97d58bfe7d63610a66f2534e1f344 has inconsistent version: filename has '3.1.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-3.0.0.tar.gz (258 kB)
     |████████████████████████████████| 258 kB 1.9 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/2c/af/c09d1dc6d4c74e73dfc7a05419336d38c6c8a63075477509946ade9712e9/cosmic-popsynth-3.0.0.tar.gz#sha256=ab65167e258bf69706fcbe0b315e7335a5d6c5865012fdae4bd0a5f8b7da1608 (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/2c/af/c09d1dc6d4c74e73dfc7a05419336d38c6c8a63075477509946ade9712e9/cosmic-popsynth-3.0.0.tar.gz#sha256=ab65167e258bf69706fcbe0b315e7335a5d6c5865012fdae4bd0a5f8b7da1608 has inconsistent version: filename has '3.0.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-2.0.0.tar.gz (236 kB)
     |████████████████████████████████| 236 kB 22.5 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/77/cb/7bf7dd219cbb2ca2d4ef8a80166353b42d5b58bf4dadac6145e546c30c41/cosmic-popsynth-2.0.0.tar.gz#sha256=8c21e1b85e493a378c2f278393dfa38b4db3d3994746289e0f5da4c5d95c7e4e (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/77/cb/7bf7dd219cbb2ca2d4ef8a80166353b42d5b58bf4dadac6145e546c30c41/cosmic-popsynth-2.0.0.tar.gz#sha256=8c21e1b85e493a378c2f278393dfa38b4db3d3994746289e0f5da4c5d95c7e4e has inconsistent version: filename has '2.0.0', but metadata has '0+unknown'
  Downloading cosmic-popsynth-1.0.1.tar.gz (243 kB)
     |████████████████████████████████| 243 kB 13.3 MB/s 
WARNING: Discarding https://files.pythonhosted.org/packages/e2/dc/638bd0d91104727edeee41b21e64e50b4b00bf71711eebfaf9c7afe7772e/cosmic-popsynth-1.0.1.tar.gz#sha256=3380e1583271d8eacda96b6d11c33aa5af3b5f48d9a485b2e49e6a6463f26db2 (from https://pypi.org/simple/cosmic-popsynth/) (requires-python:>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, <4). Requested cosmic-popsynth from https://files.pythonhosted.org/packages/e2/dc/638bd0d91104727edeee41b21e64e50b4b00bf71711eebfaf9c7afe7772e/cosmic-popsynth-1.0.1.tar.gz#sha256=3380e1583271d8eacda96b6d11c33aa5af3b5f48d9a485b2e49e6a6463f26db2 has inconsistent version: filename has '1.0.1', but metadata has '0+unknown'
ERROR: Could not find a version that satisfies the requirement cosmic-popsynth (from versions: 1.0.1, 2.0.0, 3.0.0b0, 3.0.0, 3.1.0, 3.2.0, 3.3.0, 3.4.0)
ERROR: No matching distribution found for cosmic-popsynth
````

For now the fix is to use:

````
pip install --use-deprecated legacy-resolver cosmic-popsynth
````
Which you may want to document.

Seems to be a pip issues with inconsistent metadata. But I haven't found yet what would need to be fixed.

https://stackoverflow.com/questions/67074684/pip-has-problems-with-metadata

**Comment by katiebreivik** [comment]
Thanks for bringing this up @rjfarmer! I'll have a closer look at it

**Comment by katiebreivik** [comment]
@rjfarmer  -- after many months! I think this is fixed from our 3.4.5 release. Please re-open if there is a problem!
